### PR TITLE
Define custom template sections for details pages

### PR DESCRIPTION
**Pull request by dtrucs**
Allows administrators to define custom sections for details pages (trek; touristic content; touristic event; outdoor site and outdoor courses).

Ideal for displaying certain external widgets.

Fix: https://github.com/GeotrekCE/Geotrek-rando-v3/issues/588#issuecomment-1113207825
Allows: https://github.com/GeotrekCE/Geotrek-rando-v3/issues/914 

:warning: The `enableMeteoWidget` in `global.json` is no longer supported, read the doc to see how to deactivate this widget on the details page

### CODE REVIEW

File: frontend/src/components/pages/details/Details.tsx [hunk 1]
```diff
@@ -569,6 +544,30 @@ export const DetailsUIWithoutContext: React.FC<Props> = ({ slug, parentId, langu
...
+                          <DetailsSection
+                            htmlId={`details_${section.name}`}
+                            titleId={`details.${section.name}`}
+                            className={marginDetailsChild}
+                          >
+                            <HtmlParser
+                              template={section.template}
+                              id={details.id.toString()}
+                              type="trek"
+                              cityCode={details.cities_raw[0]}
```
Comment (Bo-Duke) [inline frontend/src/components/pages/details/Details.tsx:564]: On the deleted lines, there's a `details.cities_raw?.[0]` but not here, `cities_raw` being undefined is not possible anymore?

Comment (Bo-Duke) [inline frontend/src/components/pages/details/Details.tsx:564]: On the deleted lines, there's a details.cities_raw?.[0] but not here, cities_raw being undefined is not possible anymore?

**Review by Bo-Duke — APPROVED** [review]

File: frontend/src/components/pages/details/Details.tsx (hunk 1 above)
Comment (dtrucs) [reply frontend/src/components/pages/details/Details.tsx:564]: Great point!
`cities_raw` can't be undefined but the array can be empty and i don't handle this case.

If the custom template is city-dependent and this one returns `undefined`, it's probably necessary to not display this section at all (e.g. a forecast widget).

I'll have to think about it before going any further.

Comment (dtrucs) [reply frontend/src/components/pages/details/Details.tsx:574]: The [last commit](https://github.com/GeotrekCE/Geotrek-rando-v3/pull/988/commits/e52fb05962643297490efa682bf44dd27b3a2469) allows to check is all used variables are defined to display the custom section.

**Review by Bo-Duke — APPROVED** [review]

**Pull request merged**